src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Generator.cs

```diff
@@ -27,30 +29,46 @@ namespace Microsoft.Sbom.Parsers.Spdx30SbomParser;
 /// </summary>
 public class Generator : IManifestGenerator
 {
-    private static readonly Dictionary<AlgorithmName, HashAlgorithm> AlgorithmMap = new()
-    {
-        { AlgorithmName.SHA1, HashAlgorithm.sha1 },
-        { AlgorithmName.SHA256, HashAlgorithm.sha256 },
-        { AlgorithmName.SHA512, HashAlgorithm.sha512 },
-        { AlgorithmName.MD5, HashAlgorithm.md5 }
-    };
-
     public AlgorithmName[] RequiredHashAlgorithms => new[] { AlgorithmName.SHA256, AlgorithmName.SHA1 };
 
     public string Version { get; set; } = string.Join("-", Constants.SPDXName, Constants.SPDXVersion);
 
-    string IManifestGenerator.FilesArrayHeaderName => throw new NotSupportedException();
+    private JsonSerializerOptions serializerOptions = new JsonSerializerOptions
+    {
+        Converters = { new ElementSerializer() },
+    };
 
-    string IManifestGenerator.PackagesArrayHeaderName => throw new NotSupportedException();
+    public string FilesArrayHeaderName
+    {
+        get
+        {
+            throw new NotSupportedException("Files array not supported for SBOMs generated with SPDX 3.0.");
```

Comment:
This is a perfect example of where to use a Try pattern, especially since the calling code is already using a Try pattern. The `IManifestGenerator` isn't public, so we can safely refactor this in a subsequent PR

Reply:
Will take up this change in a separate PR as suggested